lib/apns/payload.ex

```diff
@@ -0,0 +1,121 @@
+defmodule APNS.Payload do
+  def build_json(message, limit) do
+    payload = %{aps: %{}}
+
+    if message.sound do
+      payload = put_in(payload[:aps][:sound], message.sound)
+    end
+
+    if message.category != nil do
+      payload = put_in(payload[:aps][:category], message.category)
+    end
+
+    if message.badge != nil do
+      payload = put_in(payload[:aps][:badge], message.badge)
+    end
+
+    if message.content_available != nil do
+      payload = put_in(payload[:aps][:'content-available'], message.content_available)
+    end
+
+    if message.extra != [] do
+      payload = Map.merge(payload, message.extra)
+    end
+
+    if is_binary(message.alert) do
+      payload = put_in(payload[:aps][:alert], message.alert)
+    else
+      payload = put_in(payload[:aps][:alert], format_loc(message.alert))
+    end
```

Comment:
Refactor to avoid imperative cases. Consider trying to use more pattern matching.